### PR TITLE
ci: Fix ifs for jobs and rename master to main

### PR DESCRIPTION
**Pull request by wenzeslaus**
* Reference to branches (heads), not tags, to state the branch condition positively.
* Remove comparison to boolean in the condition for better readability.
* Fix comparison for master/main (use equal operator, startsWith may match too much).
* Update for rename from master to main for the default branch.

### CODE REVIEW

**Comment by wenzeslaus** [comment]
Please review @mmacata.

**Comment by wenzeslaus** [comment]
One change was not necessary. I missed it in the documentation. Reverted and the description updated accordingly.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -32,13 +32,13 @@ env:
...
   # Only run for push to configured branches, do not run for releases.
-  # Take care of different os. For master branch, created tags are:
+  # Take care of different os. For main branch, created tags are:
   # latest-alpine, latest-debian, latest-ubuntu
   # For releasebranch_7_8, created tags are:
   # stable-alpine, stable-debian, stable-ubuntu
   docker-branch-os-matrix:
     name: build and push ${{ matrix.os }} for branch
-    if: startsWith(github.ref, 'refs/tags/') != true
+    if: startsWith(github.ref, 'refs/heads/') != true
```
Comment (mmacata) [inline .github/workflows/docker.yml:41]: This job is meant to run for everything except tags. Now it prevents the run for branches which is needed here. I would agree if the negotioation is removed (`!= true`) then maybe it is more clear. The check for `refs/heads/` should be sufficient, I don't think any other ref type is relevant here.

**Review by mmacata — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (wenzeslaus) [reply .github/workflows/docker.yml]: Right. I misunderstood the condition. As far as I can tell from Git and GitHub doc, `refs/heads/` is branches, so exactly what we want to query for. Keeping `startsWith...heads`, dropping the `!= true` part.

**Comment by wenzeslaus** [comment]
Thanks @mmacata. Code and PR description updated. Please, review again if you can.

**Review by mmacata — APPROVED** [review]
Fine for me to merge. Thank you!

**Pull request merged**